cms_frontend/src/pages/Dashboard.tsx

```diff
@@ -41,45 +38,146 @@ const Dashboard: React.FC = () => {
 
   // Moves our current directory up (analogous to `cd ..`)
   const toParent = () => {
-    const parent = getParent() as FolderName;
+    const parent = getParent();
     setDir(parent);
+    setFolder(Files.get(parent) as FileFormat[]);
   }
 
   // Checks if our current directory has a parent directory
   const hasParent = () => {
     const parent = getParent();
+    let found = false;
+
+    Files.forEach((_, key) => {
+      if (key === parent) {
+        found = true;
+      }
+    });
 
-    for (const key in Files) {
-      if (parent === key) {
+    return found;
+  }
+
+  // Checks if a file/folder name already exists in our current directory
+  const containsFilename = (name: string, type: string) => {
+    for (const item of folder) {
+      if (item.type === type && name === item.filename) {
         return true;
       }
     }
 
     return false;
   }
 
+  // Updates our current folder state, also emitting it to the backend
+  // TODO: change to an API call once that's up
+  const updateFolder = (updated: FileFormat[]) => {
+    setFolder(updated);
+    Files.set(dir, updated);
+  }
+
+  // Finds the next available folder name
+  const newFolderName = () => {
+    let index = 0;
+    let folder_name = "New Folder";
+
+    while (containsFilename(folder_name, "folder")) {
+      index++;

```

Comment:
I'm quite confused as to what this does, and whether or not it will end up in an infinite loop

Reply:
This generates a name for a new folder ("New Folder", "New Folder (1)", "New Folder (2)") and uses the first available name, shouldn't infinite loop since a directory only has a finite number of files so we'll eventually find a suitable name

Reply:
right Okay i didnt see where folder_name was coming from, all goods looks good then!!